### PR TITLE
sql: hide enable_drop_enum_value session/cluster setting 

### LINKED ISSUE #61594: sql: move `ALTER TYPE ... DROP VALUE ...` out of alpha
**Is your feature request related to a problem? Please describe.**

Currently, `ALTER TYPE ... DROP VALUE` is supported as an alpha feature and to that end is gated behind the `enable_drop_enum_value` SESSION setting (or the corresponding `sql.defaults.drop_enum_value.enabled` CLUSTER setting. The main reason behind this gating this functionality behind a feature flag are the limitations described in https://github.com/cockroachdb/cockroach/issues/59807. Once those limitations are resolved, we should move this feature out of alpha. 

Followup as a result of #61519

Epic CRDB-2451

### PR DESCRIPTION
**Pull request by rafiss**
fixes https://github.com/cockroachdb/cockroach/issues/61594 

Release note (sql change): The enable_drop_enum_value session variable
has been removed, along with the corresponding cluter setting. The
functionality of being able to drop enum values is now enabled
automatically. Queries that refer to the session/cluster setting will
still work but will have no effect.

### CODE REVIEW

**Review by otan — APPROVED** [review]

File: pkg/sql/sessiondatapb/local_only_session_data.proto [hunk 1]
```diff
@@ -97,8 +97,6 @@ message LocalOnlySessionData {
   // ImplicitPartitioningEnabled indicates whether implicit column partitioning
   // can be created.
   bool implicit_column_partitioning_enabled = 22;
-  // DropEnumValueEnabled indicates whether enum values can be dropped.
-  bool drop_enum_value_enabled = 23;
```
Comment (postamar) [inline pkg/sql/sessiondatapb/local_only_session_data.proto:101]: Shouldn't this be replaced with `reserved 23` until 22.1 is released?

**Review by postamar — COMMENTED** [review]
I just have one small comment to make, otherwise LGTM.

File: pkg/sql/sessiondatapb/local_only_session_data.proto (hunk 1 above)
Comment (rafiss) [reply pkg/sql/sessiondatapb/local_only_session_data.proto:101]: nice catch, i wasn't familiar with the right way to do this

**Comment by rafiss** [comment]
tftrs!

bors r=otan,postamar

**Review by RaduBerinde — COMMENTED** [review]
[nit] Worth clarifying in the title and commit description that we are removing the cluster setting and leaving the session setting as a dummy (to prevent existing scripts from failing). It's helpful in case others look summarily at PRs like this to establish precedent when it comes to removing settings. BTW we may want a separate list of retired session settings (for which we could tolerate any value) to make the code cleaner.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70369#-:-Mk3RQ-I0K07fDauVnmq:b-bolteb)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @arulajmani)

**Comment by rafiss** [comment]
sure, i will clarify that

bors r-

**Comment by rafiss** [comment]
bors r=otan,postamar

**Pull request merged**